atomic_reactor/plugins/post_import_image.py

```diff
@@ -88,19 +88,40 @@ def run(self):
                           self.docker_image_repo)
 
             # Tags are imported automatically on creation
-            osbs.create_image_stream(self.imagestream, self.docker_image_repo,
-                                     **kwargs)
-        else:
-            self.log.info("Importing new tags for %s", self.imagestream)
-
-            attempts = 0
-            while not osbs.import_image(self.imagestream):
-                attempts += 1
-
-                if attempts >= self.import_attempts:
-                    msg = "Failed to import new tags for %s"
-                    raise RuntimeError(msg % self.imagestream)
-
-                self.log.info("no new tags, will retry after %d seconds (%d/%d)",
-                              self.retry_delay, attempts, self.import_attempts)
-                sleep(self.retry_delay)
+            imagestream = osbs.create_image_stream(self.imagestream,
+                                                   self.docker_image_repo,
+                                                   **kwargs)
+        self.log.info("Importing new tags for %s", self.imagestream)
+
+        primaries = None
+        try:
+            primaries = self.workflow.build_result.annotations['repositories']['primary']
+        except TypeError:
+            pass
+        except KeyError:
+            pass
+        if primaries:
+            for s in primaries:
+                tag_image_name = ImageName.parse(s)
+                tag = tag_image_name.tag
+                failures = False
+                try:
+                    osbs.ensure_image_stream_tag(imagestream.json(), tag)
+                    self.log.info("Imported ImageStreamTag: " + str(tag))
```

Comment:
Tiny nitpick: might be nicer to use `self.log.info("Imported ImageStreamTag: %s", tag)`, just because then the log record has an `args`.